migrations/2025-04-25-183519_show_downvotes_for_others_only/up.sql

```diff
@@ -0,0 +1,25 @@
+-- This changes the local_user.show_downvotes column to an enum,
+-- which by default hides downvotes to your own content.
+CREATE TYPE vote_show_enum AS ENUM (
+    'ShowForOthers',
+    'Show',
+    'Hide'
+);
+
+ALTER TABLE local_user
+    ALTER COLUMN show_downvotes DROP DEFAULT;
+
+ALTER TABLE local_user
+    ALTER COLUMN show_downvotes TYPE vote_show_enum
+    USING
+        CASE show_downvotes
+        WHEN FALSE THEN
+            'Hide'
+        ELSE
+            'ShowForOthers'
+        END::vote_show_enum;
+
+-- Make ShowForOthers the default
+ALTER TABLE local_user
+    ALTER COLUMN show_downvotes SET DEFAULT 'ShowForOthers';
```

Comment:
I would leave the default as is, most people seem fine with it.

Reply:
I would like `ShowForOthers` to be the new default, for the reasons in [the issue body](https://github.com/LemmyNet/lemmy/issues/5639#issue-3015440290). ppl can always change it at any time.

Reply:
On the other hand it would be very confusing to make a post which receives lots of downvotes, but not even being aware of them.

Reply:
The positives for mental health from not seeing those by default outweigh the negatives IMO. Again ppl can always re-enable this at any time.

Reply:
Since this seems controversial enough, I'll change the default to `Show`